### PR TITLE
Save to settings

### PR DESCRIPTION
**Pull request by casperstorm**
This adds a `config::Channel` struct where we can add different settings which is channel specific. Editing these values will save the config so its persisted.

I moved `user_color` to be a global setting, and removed it from module header. 
Does not make sense to set it on a per buffer basis imo.

### CODE REVIEW

File: data/src/channel.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct Config {
+    pub show_user_list: bool,
```
Comment (casperstorm) [inline data/src/channel.rs:5]: Do we have a better name for this? 🤔

File: data/src/config.rs [hunk 2]
```diff
@@ -7,13 +7,46 @@ use serde::{Deserialize, Serialize};
...
 
 #[derive(Debug, Clone, Default, Deserialize, Serialize)]
 pub struct Config {
     #[serde(default)]
     pub palette: Palette,
     pub servers: BTreeMap<String, server::Config>,
+    #[serde(default)]
+    pub channels: BTreeMap<String, BTreeMap<String, channel::Config>>,
+    #[serde(default)]
+    pub user_color: UserColor,
```
Comment (casperstorm) [inline data/src/config.rs:20]: Suggestion for a better name?

File: src/buffer/channel.rs [hunk 3]
```diff
@@ -64,7 +63,7 @@ pub fn view<'a>(
     });
 
     // TODO: Maybe we should show it to the right instead of left.
```
Comment (casperstorm) [inline src/buffer/channel.rs:65]: Not using position in `User` position user sidebar. Will add that later. We have the information now, which was the goal.

**Review by tarkah — APPROVED** [review]
LGTM!

**Pull request merged**